CMakeLists.txt

```diff
@@ -12,7 +12,13 @@
 #
 ##===----------------------------------------------------------------------===##
 
-cmake_minimum_required(VERSION 3.4.3)
+if (CMAKE_HOST_WIN32)
```

Comment:
I've noticed that we use the variables `CMAKE_HOST_WIN32` or `WIN32` pretty much interchangeably. Do you think we should converge on a single one?

Reply:
They have different meanings, `CMAKE_HOST_WIN32` is for the host system, where `WIN32` is the for the target (cross compilation).  

I think for these purposes, it is the host system that we are interested in, because it is about getting things to properly compile using the Windows versions of CMake and the intel compilers or Linux versions of the same. 

Taking a quick look around the rest of the CMake files:

- I just see one usage of `WIN32` in the `test/CMakeLists`, with the recent work on checked iterators, and I think that is appropriate.  
- A few others in examples which I think are OK. 
- Finally there is one in the generation script for the CMake config.  I think that one is also appropriate as `WIN32`.  

To me they seem appropriate but I may be mistaken somewhere.